src/Dhall/Core.hs

```diff
@@ -494,6 +496,11 @@ instance IsString (Chunks s a) where
     case the corresponding builder.
 -}
 
+data DhallAnn = Keyword | Variable | Punctuation | Label | Literal
```

Comment:
Perhaps add a separate `Reserved` category for built-in sorts/kinds/types/values (i.e. `Type`/`Kind`/`Bool`/`True`) instead of classifying them as `Keyword`s.  I normally associate "keyword" with things like `if`/`then`/`else` but not built-in functions or values like `True` or `Natural/even`

Also, perhaps rename `Punctuation` to `Operator` since it looks like you treat record/union/list literals as `Literal`s (which I think makes sense)

Reply:
I think `Reserved` (or `Builtin`?) is good for builtin values, separating them from keywords which are syntactic. I think I will rename `Punctuation` to `Syntax` (parens, braces, commas, etc). Dhall's operators would be `Reserved`/`Builtin` (here my preference is `Builtin`).